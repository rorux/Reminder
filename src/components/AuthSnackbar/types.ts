export interface IAuthSnackbarProps {
  message: string;
  clearError: () => void;
}
