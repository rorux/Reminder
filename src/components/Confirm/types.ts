import React from 'react';

export type TConfirmProps = {
  open: boolean;
  setOpen: React.Dispatch<React.SetStateAction<boolean>>;
  handler: (event: React.MouseEvent<HTMLElement>, id: string) => void;
  id: string;
};
