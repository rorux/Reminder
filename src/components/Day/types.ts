import { TRecord } from '@store/records/types';

export type TDayProps = {
  number: null | number;
  records: Array<TRecord>;
};

export type TStylesDay = {
  backgroundColor?: string;
  cursor?: 'pointer';
};
