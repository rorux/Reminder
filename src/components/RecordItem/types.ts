import React from 'react';
import { TRecordAccordion } from '@components/RecordsAccordion/types';

export type TRecordItemProps = {
  record: TRecordAccordion;
  handleChange: (panel: string) => (event: React.SyntheticEvent, isExpanded: boolean) => void;
  expanded: string | false;
  handleDelete: (event: React.MouseEvent<HTMLElement>, id: string) => void;
};
