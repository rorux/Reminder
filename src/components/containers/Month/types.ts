import { TRecord } from '@store/records/types';

export type TMonthProps = {
  year: number;
  month: number;
  recordList: Array<TRecord>;
};
