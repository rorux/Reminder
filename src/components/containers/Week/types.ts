import { TRecord } from '@store/records/types';

export type TWeekProps = {
  weekArray: Array<null | number>;
  recordList: Array<TRecord>;
};
