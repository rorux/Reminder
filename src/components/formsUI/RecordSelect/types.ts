export type TSelectParams = {
  value: string | number;
  name: string | number;
};

export type TSelectProps = {
  name: string;
  params: Array<TSelectParams>;
  setter: any;
  value?: any;
};
