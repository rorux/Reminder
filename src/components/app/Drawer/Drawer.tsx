import React, { useCallback } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { styled } from '@mui/material/styles';
import Toolbar from '@mui/material/Toolbar';
import IconButton from '@mui/material/IconButton';
import MuiDrawer from '@mui/material/Drawer';
import ChevronLeftIcon from '@mui/icons-material/ChevronLeft';
import List from '@mui/material/List';
import Divider from '@mui/material/Divider';
import { appSelector } from '@store/app/selectors';
import { drawerToggleAction } from '@store/app/actions';
import Menu from '../Menu';

const DrawerStyled = styled(MuiDrawer, { shouldForwardProp: (prop) => prop !== 'open' })(
  ({ theme, open }) => ({
    '& .MuiDrawer-paper': {
      position: 'relative',
      whiteSpace: 'nowrap',
      width: 240,
      transition: theme.transitions.create('width', {
        easing: theme.transitions.easing.sharp,
        duration: theme.transitions.duration.enteringScreen,
      }),
      boxSizing: 'border-box',
      ...(!open && {
        overflowX: 'hidden',
        transition: theme.transitions.create('width', {
          easing: theme.transitions.easing.sharp,
          duration: theme.transitions.duration.leavingScreen,
        }),
        width: theme.spacing(7),
        [theme.breakpoints.up('sm')]: {
          width: theme.spacing(9),
        },
      }),
    },
  })
);

export default function Drawer() {
  const { isDrawerOpen } = useSelector(appSelector);
  const dispatch = useDispatch();

  const toggleDrawer = useCallback(() => {
    dispatch(drawerToggleAction());
  }, [dispatch]);

  return (
    <DrawerStyled variant="permanent" open={isDrawerOpen}>
      <Toolbar
        sx={{
          display: 'flex',
          alignItems: 'center',
          justifyContent: 'flex-end',
          px: [1],
        }}
      >
        <IconButton onClick={toggleDrawer}>
          <ChevronLeftIcon />
        </IconButton>
      </Toolbar>
      <Divider />
      <List>
        <Menu />
      </List>
    </DrawerStyled>
  );
}
