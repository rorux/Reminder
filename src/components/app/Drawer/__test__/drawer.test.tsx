import React from 'react';
import * as redux from 'react-redux';
import IconButton from '@mui/material/IconButton';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import Enzyme, { shallow } from 'enzyme';
import Drawer, { DrawerStyled } from '../Drawer';
import { AppActionTypes } from '@store/app/types';

Enzyme.configure({ adapter: new Adapter() });

describe('Drawer component', () => {
  let component: Enzyme.ShallowWrapper;
  let spyOnUseSelector;
  let spyOnUseDispatch;
  let mockDispatch: jest.Mock;

  beforeEach(() => {
    spyOnUseSelector = jest.spyOn(redux, 'useSelector');
    spyOnUseSelector.mockReturnValue({
      isDrawerOpen: false,
    });
    spyOnUseDispatch = jest.spyOn(redux, 'useDispatch');
    mockDispatch = jest.fn();
    spyOnUseDispatch.mockReturnValue(mockDispatch);
    component = shallow(<Drawer />);
  });

  afterEach(() => {
    jest.restoreAllMocks();
  });

  it('Should render Drawer component', () => {
    expect(component).toMatchSnapshot();
  });

  it('Should dispatch drawerToggleAction when clicked toggle', () => {
    // @ts-ignore
    component.find(IconButton).props().onClick();
    expect(mockDispatch.mock.calls[0][0]).toEqual({
      type: AppActionTypes.DRAWER_TOGGLE,
    });
  });
});

describe('DrawerStyled component', () => {
  it('Should render DrawerStyled component', () => {
    expect(shallow(<DrawerStyled />)).toMatchSnapshot();
  });
});
