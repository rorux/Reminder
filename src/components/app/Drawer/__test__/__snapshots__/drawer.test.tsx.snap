// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Drawer component Should render Drawer component 1`] = `
<Styled(ForwardRef(Drawer))
  open={false}
  variant="permanent"
>
  <ForwardRef(Toolbar)
    sx={
      Object {
        "alignItems": "center",
        "display": "flex",
        "justifyContent": "flex-end",
        "px": Array [
          1,
        ],
      }
    }
  >
    <ForwardRef(IconButton)
      onClick={[Function]}
    >
      <Memo(ForwardRef(ChevronLeftIcon)) />
    </ForwardRef(IconButton)>
  </ForwardRef(Toolbar)>
  <ForwardRef(Divider) />
  <ForwardRef(List)>
    <Menu />
  </ForwardRef(List)>
</Styled(ForwardRef(Drawer))>
`;

exports[`DrawerStyled component Should render DrawerStyled component 1`] = `
<ContextProvider
  value={
    Object {
      "insert": [Function],
      "inserted": Object {
        "1x1jwk9": true,
      },
      "key": "css",
      "nonce": undefined,
      "registered": Object {
        "css-1x1jwk9": "& .MuiDrawer-paper{position:relative;white-space:nowrap;width:56px;transition:width 195ms cubic-bezier(0.4, 0, 0.6, 1) 0ms;box-sizing:border-box;overflow-x:hidden;@media (min-width:600px){width:72px;}}",
      },
      "sheet": StyleSheet {
        "_insertTag": [Function],
        "before": null,
        "container": undefined,
        "ctr": 0,
        "insertionPoint": undefined,
        "isSpeedy": false,
        "key": "css",
        "nonce": undefined,
        "prepend": undefined,
        "tags": Array [],
      },
    }
  }
>
  <style
    dangerouslySetInnerHTML={
      Object {
        "__html": ".css-1x1jwk9 .MuiDrawer-paper{position:relative;white-space:nowrap;width:56px;-webkit-transition:width 195ms cubic-bezier(0.4, 0, 0.6, 1) 0ms;transition:width 195ms cubic-bezier(0.4, 0, 0.6, 1) 0ms;box-sizing:border-box;overflow-x:hidden;}@media (min-width:600px){.css-1x1jwk9 .MuiDrawer-paper{width:72px;}}",
      }
    }
    data-emotion="css 1x1jwk9"
  />
  <ForwardRef(Drawer)
    className="css-1x1jwk9"
  />
</ContextProvider>
`;
